### DISCUSSION TITLE
Trouble install brew on Mac M1 (Xcode 12.5)

### DISCUSSION THREAD

**Discussion by Chnuts** [Homebrew/discussions#1379] [Getting started]
Hello,

I have trouble with brew install, I'm running following command : 
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` 
and I have this response :
`
==> Checking for 'sudo' access (which may request your password).
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew
==> The following existing directories will be made writable by user only:
/opt/homebrew/share/zsh
/opt/homebrew/share/zsh/site-functions

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/chmod u+rwx /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions
==> /usr/bin/sudo /bin/chmod 755 /opt/homebrew/share/zsh /opt/homebrew/share/zsh/site-functions
==> /usr/bin/sudo /usr/sbin/chown -R Francois:admin /opt/homebrew
==> Downloading and installing Homebrew...
/bin/bash: line 156: /usr/local/bin/git: Bad CPU type in executable
Failed during: git init -q
`
I have Xcode 12.5 and up to date with Command line tools (xcode-select --install ==>
xcode-select: error: command line tools are already installed, use "Software Update" to install updates)

Any idea?

Thanks

**Comment by carlocab** [comment]
What's the output of
```
xcode-select --print-path
/usr/bin/git --version
ls -l /Library/Developer/CommandLineTools/SDKs
```
?

**Comment by gromgit** [comment]
> ```
> /bin/bash: line 156: /usr/local/bin/git: Bad CPU type in executable
> ```

This suggests that you already have an _Intel_ (via Rosetta 2) Homebrew installation, and you installed `git` there. Since you have `/usr/local/bin` in your `PATH`, the installer is using that `git`...and failing.

Try this instead:
1. Open a new Terminal session.
2. Remove `/usr/local/bin` from `PATH` for this session: `PATH=${PATH/\/usr\/local\/bin:}`
3. Run `which git curl` in this session. Confirm that you get no errors _and_ neither path reported begins with `/usr/local/bin`
4. Run the installer in this session.

**Reply by carlocab** [reply]
Surprised to see the `Bad CPU type` error though. But that might be a consequence of not having Rosetta installed. Good catch.

**Comment by Chnuts** [comment]
I resolved my problem with your help. 

I removed `/usr/local/bin` from `PATH`. After that,  run `which git curl` display following response :
```
/usr/bin/git
/usr/bin/curl
```
I was able to install homebrew successfully, however after installation, run `which git curl` display following 
```
/usr/local/bin/git
/usr/bin/curl
```
I don't understand how it's possible. Any issue in `brew` ?

**Reply by carlocab** [reply]
No. You probably restarted your terminal or reloaded your shell settings.

**Comment by Cakmakli-a** [comment]
hello,

unfortunately i have the "same" problem. installation ends with:
```
==> /usr/bin/sudo /bin/mkdir -p /opt/homebrew
==> /usr/bin/sudo /usr/sbin/chown -R myUserName:admin /opt/homebrew
==> Downloading and installing Homebrew...
/bin/bash: line 860: cd: /opt/homebrew: Permission denied
/bin/bash: line 860: return: can only `return' from a function or sourced script
/usr/.git: Operation not permitted
Failed during: git init -q
```

`which git curl` returns:
```
/usr/bin/git
/usr/bin/curl
```

`xcode-select --print-path` returns `/Applications/Xcode.app/Contents/Developer`

using `git version 2.32.0 (Apple Git-132)`

have admin rights. running MacOS 12.1 (Monterey) on Apple M1.

**Reply by gromgit** [reply]
What's the output of `/bin/ls -ldeO@ /opt/ /opt/homebrew`?

**Reply by Cakmakli-a** [reply]
```
/bin/ls -ldeO@ /opt/ /opt/homebrew
ls: /opt/homebrew: Permission denied
drw-rw-r--  2 root  wheel  hidden 64  1 Jan 20:08 /opt/
```

i delete the folder `homebrew` after failure of installation

**Reply by gromgit** [reply]
> ```
> drw-rw-r--  2 root  wheel  hidden 64  1 Jan 20:08 /opt/
> ```

Your `/opt` doesn't have execute permissions at all, so no user (including root) can descend into that directory.

Did you `sudo chmod 664 /opt`? If so, `sudo chmod 755 /opt` to fix it, then try installing Homebrew again. (I don't have an M1 Mac yet, so I'm referencing the `/opt` permissions on Intel macOS.)

**Reply by Cakmakli-a** [reply]
`sudo chmod 755 /opt` does help.

thx a lot

**Comment by ashishbaberwal** [comment]
arch: posix_spawnp: /bin/bash: Bad CPU type in executable

plz help me

**Reply by Bo98** [reply]
Sounds like you are trying to run a command with `arch -x86_64` without having Rosetta 2 installed.

**Reply by ashishbaberwal** [reply]
I have m2 mac

**Comment by aaronjwood** [comment]
In case others are running into this when they have Rosetta installed and a generally clean/correct system to begin with: https://github.com/orgs/Homebrew/discussions/3385#discussioncomment-15622869
